File: dart/src/n02_test_data_generator/data_generator.dart
import 'dart:io';
import 'dart:math';

import 'package:meta/meta.dart';

import 'data_set_saver.dart';

class DataGenerator {
  @protected
  final List<double> dataSet = List.empty(growable: true);
  DataSetSaver? _saver;

  void createData(final double min, final double max) {
    final Random random = Random();
    for (int i = 0; i < 100; i++) {
      this.dataSet.add(random.nextDouble() * (max - min) + min);
    }
  }

  set saver(final DataSetSaver saver) => this._saver = saver;

  Future<void> saveData(final File file) => this._saver!.saveData(file, this.dataSet);
}

File: java/src/n02_test_data_generator/DataGenerator.java
package n02_test_data_generator;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class DataGenerator {
    protected List<Double> dataSet = new ArrayList<>();
    private DataSetSaver saver;

    public void createData(double min, double max) {
        Random random = new Random();
        for (int i = 0; i < 100; i++) {
            this.dataSet.add(random.nextDouble(min, max));
        }
    }

    public void setDataSetSaver(DataSetSaver saver) {
        this.saver = saver;
    }

    void saveData(File file) {
        this.saver.saveData(file, this.dataSet);
    }
}
